editions/tw5.com/tiddlers/macros/examples/toolbar Macro (Examples).tid

```diff
@@ -0,0 +1,26 @@
+created: 20210623113149086
+modified: 20210628094054417
+tags: [[Macro Examples]] [[toolbar Macros]]
+title: toolbar Macro (Examples)
+
+This example create a replica of ~TiddlyWiki ~PageControls:
+
+<$macrocall $name=".example" n="1" eg="""<div class="tc-page-controls">
+<<toolbar "$:/tags/PageControls" "$:/config/PageControlButtons/Visibility/">>
+</div>"""/>
+
+
+This example create a replica of ~TiddlyWiki ~PageControls with class "//tc-btn-rounded//", without change the general style for toolbar buttons:
+
+<$macrocall $name=".example" n="2" eg="""<div class="tc-page-controls">
+<<toolbar "$:/tags/PageControls" "$:/config/PageControlButtons/Visibility/" "tc-btn-rounded">>
```

Comment:
IMO 1 of the examples should have "named parameters" eg: `tag:"$:/tags/PageControls"` and so on. 

_And_ the example for the `template` parameter is missing. ... IMO it would be interesting, to know, how it is meant to be used

Reply:
How you coud see i was following the doc of **toc** macro. The "named parameters" are not used there. And I think that there are multiple options to do a macro call.
- Unnamed and ordered parameters (which are used in examples)
- Named and ordered parameters (how you say it)
- Named and unordered parameters `<<toolbar config:"$:/config/PageControlButtons/Visibility/" tag:"$:/tags/PageControls">>`
- named and (un)ordered parameters mixed with unnamed and ordered parameters. `<<toolbar config:"$:/config/PageControlButtons/Visibility/" "$:/tags/PageControls">>`
But I think is better use the same syntax in the examples and maybe we can update the documentation of `Macro Call Syntax` and then we can add, in the tiddlers with examples, a link to (updated) documentation.

I agree. But I have several doubts/problems with this. I don't know how work with templates for buttons and the implementation of editor is WIP, the editor is in early step, its dropdown doesn't work. In short, I have doubts about if it is convenient create a ad hoc example and I don't know well how I can create it; and the alternative is in WIP and if it can't be confused to new users and few expert users can use it and they do it few times.

Reply:
The following code works as example, but "more action" button in this toolbar doesn't work. I have to fix it.
```
<div  class="tc-editor-toolbar">
<<toolbar "$:/tags/EditorToolbar" "$:/config/EditorToolbarButtons/Visibility/" template:"$:/core/ui/EditTemplate/body/toolbar/button">>
</div>
```